src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -522,18 +594,38 @@ private void markDuplicateFragments(final List<ReadEndsForMarkDuplicates> list,
         }
     }
 
+    // To avoid overflows or underflows
```

Comment:
// To avoid overflows or underflows when subtracting two large (positive and negative) numbers